### PR TITLE
Feature/sketch

### PR DESCRIPTION
**Pull request by Yerion**
Sketch branch is ready for review
@johanstokking: Just the code (ino and js file)
@laurensslats: Please look at the README

### CODE REVIEW

**Review by johanstokking — CHANGES_REQUESTED** [review]
- Use lowercase folder names everywhere
- Don't use big comment blocks with those `*****`
- Use a code formatter that takes care of spaces and tabs. Install a proper plugin in your editor and configure it correctly. Hint: the code snippets in the README are messy, also mixing things like `char *x` and `char* x`
- Look at the commit messages; fixup, commit amend, reword, etc, make sure it's clean
- Don't add Dutch screenshots

**Comment by johanstokking** [comment]
Reopen when it's ready to review.

Few things at first sight:
- Indentation is still not good
- Commit messages follow: `low: Description`. Please reword this branch
- Group connector instructions together, preferably not in the code but in corresponding documentation
- Don't globally declare variables when that is not necessary

**Comment by Yerion** [comment]
* Indentation is still not good
Where am I messing up? I can't see it.
* Commit messages follow: low: Description. Please reword this branch
What do you mean with: low?
* Group connector instructions together, preferably not in the code but in corresponding documentation
Do you mean comments like these: // When using the sodaq explorer connect the ESP TX to pin D0 and RX to D1?

File: Sketch/payload.js [hunk 1]
```diff
@@ -0,0 +1,57 @@
+function Decoder(bytes, port) {
+	var Bssid1 = [6];
+	var Bssid2 = [6];
+	var Bssid3 = [6];
```
Comment (johanstokking) [inline Sketch/payload.js:4]: - What is this?
- Also use lowercase variable names

File: Sketch/payload.js [hunk 2]
```diff
@@ -0,0 +1,57 @@
...
+	var i, j;
+	var bit;
+
+	var decoded = {};
+	for(i = 0, j = 0; i < 6 && j < 6; i++, j++)
```
Comment (johanstokking) [inline Sketch/payload.js:9]: Turn this into a function you call three times

File: Sketch/payload.js [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+		{
+			Bssid3[i] = pad(bit) + ":";
+		}
+		else
+		{
+			Bssid3[i] = pad(bit);
+		}
+	}
+
+	decoded.Bssid1 = Bssid1.join('') ;
```
Comment (johanstokking) [inline Sketch/payload.js:46]: Formatting

File: Sketch/payload.js [hunk 4]
```diff
@@ -0,0 +1,57 @@
...
+	decoded.Bssid2 = Bssid2.join('') ;
+	decoded.Bssid3 = Bssid3.join('') ;
+
+	return decoded;
+}
+
+
+function pad(n)
+{
+return  (n < 10 || n == 'a' || n =='b' || n == 'c' || n == 'd' || n == 'e' || n == 'f' ) ? ("0" + n) : n;
```
Comment (johanstokking) [inline Sketch/payload.js:56]: Formatting and indentation

File: Sketch/README.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+#define espSerial Serial // D0 & D1
+
+void setup()
+{
+  while (!debugSerial);
+
+  debugSerial.begin(9600);
+
+  delay(500);
+  espSerial.begin (9600); //Possible baud rates to try: 9600, 19200, 38400, 57600. Don’t forget to change the baud rate in the serial monitor accordingly
```
Comment (johanstokking) [inline Sketch/README.md:51]: Formatting and put instruction in setup doc

File: Sketch/README.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+* The payload is sent to the console and after a delay of a minute everything is repeated.
+
+### Functions
+---
+The following functions are used in the sketch: `WaitForOK`, `readLine`, `sendBssid`.
+```
+bool waitForOK(uint32_t waitTime) 
+{
+  uint32_t timeout = millis() + waitTime;
+    while (millis() < timeout)
```
Comment (johanstokking) [inline Sketch/README.md:157]: Indentation

File: Sketch/README.md [hunk 7]
```diff
@@ -0,0 +1,208 @@
...
+---
+The following functions are used in the sketch: `WaitForOK`, `readLine`, `sendBssid`.
+```
+bool waitForOK(uint32_t waitTime) 
+{
+  uint32_t timeout = millis() + waitTime;
+    while (millis() < timeout) 
+    {
+      uint8_t l = espSerial.readBytesUntil('\n', buffer, SF_ESP8266_BUFFER_SIZE);
+        if (l > 0 && strncmp("OK", buffer, 2) == 0)
```
Comment (johanstokking) [inline Sketch/README.md:160]: Indentation, many other occurrences in this file

File: Sketch/Explorer_ESP_WiFi-localization_v1.2.ino [hunk 8]
```diff
@@ -0,0 +1,170 @@
...
+			{
+				aps[bssidNumber][bssidByte] = HEX_PAIR_TO_BYTE(b[0], b[1]);
+			}
+			bssidNumber++;
+		}
+	}
+
+	sendBssid();
+	exit:
+	debugSerial.println("Restart the loop");
```
Comment (johanstokking) [inline Sketch/Explorer_ESP_WiFi-localization_v1.2.ino:123]: Indentation

File: Sketch/Explorer_ESP_WiFi-localization_v1.2.ino [hunk 9]
```diff
@@ -0,0 +1,170 @@
...
+			break;
+		}
+		if (strncmp("+CWLAP", line, 2) != 0)
+		{
+			delay(1);
+			debugSerial.println(F("No Bssid"));
+			continue;
+		}
+
+// Break the accesspoint in tokens, to get the bssid out.
```
Comment (johanstokking) [inline Sketch/Explorer_ESP_WiFi-localization_v1.2.ino:99]: Indentation

**Review by johanstokking — COMMENTED** [review]
See below for some examples.

With commit messages, I mean that the first part is lowercase and refers to something, for example `sketch`, `doc`, etc. No branch names, no uppercase, no inconsistencies.

Indeed those comments.

**Pull request closed without merging**